compiler/src/diagnostics/comments.re

```diff
@@ -374,3 +374,49 @@ module Doc = {
     List.rev(section_comments^);
   };
 };
+
+let print_comment = (comment: Parsetree.comment) => {
+  let endloc =
+    switch (comment) {
+    | Line(cmt)
+    | Block(cmt)
+    | Doc(cmt)
+    | Shebang(cmt) => cmt.cmt_loc.loc_end
+    };
+
+  let startloc =
+    switch (comment) {
+    | Line(cmt)
+    | Block(cmt)
+    | Doc(cmt)
+    | Shebang(cmt) => cmt.cmt_loc.loc_start
+    };
+
+  let (_file, stmtstartline, startchar, _sbol) =
+    Locations.get_raw_pos_info(startloc);
+  let (_file, stmtendline, endchar, _sbol) =
+    Locations.get_raw_pos_info(endloc);
+
+  Printf.printf(
+    "%d:%d,%d:%d -",
+    stmtstartline,
+    startchar,
+    stmtendline,
+    endchar,
+  );
+
+  switch (comment) {
+  | Line(cmt)
+  | Block(cmt)
+  | Doc(cmt)
+  | Shebang(cmt) => print_endline(cmt.cmt_source)
+  };
+};
+
+let cmt_text = (comment: Parsetree.comment) =>
```

Comment:
I'd like to see this named like `get_source` in the future or something, but I'm not going to hold my approval on that.

Reply:
Good call, done that.